.github/workflows/perf-test-artifact.yaml

```diff
@@ -0,0 +1,365 @@
+        uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # v3.3.2
+        with:
+          path: ~/artifact-build
+          key: node-build-artifacts-${{ needs.validate.outputs.version }}-${{ github.sha }}
+
+      - name: Gradle Assemble
+        id: gradle-build
+        uses: gradle/gradle-build-action@842c587ad8aa4c68eeba24c396e15af4c2e9f30a # v2.9.0
+        with:
+          gradle-version: ${{ inputs.gradle-version }}
+          arguments: assemble --scan
+
+      - name: Gradle Version Summary
+        uses: gradle/gradle-build-action@842c587ad8aa4c68eeba24c396e15af4c2e9f30a # v2.9.0
+        with:
+          gradle-version: ${{ inputs.gradle-version }}
+          arguments: githubVersionSummary --scan
+
+      - name: Stage Artifact Build Folder
+        id: artifact-staging
+        run: |
+          BUILD_BASE_DIR="${HOME}/artifact-build"
+          mkdir -p "${BUILD_BASE_DIR}/data/lib"
+          mkdir -p "${BUILD_BASE_DIR}/data/apps"
+
+          cp -f hedera-node/data/lib/*.jar "${BUILD_BASE_DIR}/data/lib"
+          cp -f hedera-node/data/apps/*.jar "${BUILD_BASE_DIR}/data/apps"
+          cp -f hedera-node/configuration/update/immediate.sh "${BUILD_BASE_DIR}"
+          cp -f hedera-node/configuration/update/during-freeze.sh "${BUILD_BASE_DIR}"
+
+          echo "folder=${BUILD_BASE_DIR}" >>"${GITHUB_OUTPUT}"
+
+      - name: Write Artifact Version Descriptor
+        working-directory: ${{ steps.artifact-staging.outputs.folder }}
+        run: |
+          printf "VERSION=%s\nCOMMIT=%s\nDATE=%s" "${{ needs.validate.outputs.version }}" "${{ needs.validate.outputs.commit-id-short }}" "$(date -u)" | tee VERSION
+
+      - name: Create Artifact Archive
+        id: artifact-release
+        working-directory: ${{ steps.artifact-staging.outputs.folder }}
+        run: |
+          POLICY="${{ inputs.version-policy }}"
+          ARTIFACT_BASE_DIR="${HOME}/artifact-release"
+          mkdir -p "${ARTIFACT_BASE_DIR}"
+
+          if [[ "${POLICY}" == "branch-commit"  ]]; then
+            ARTIFACT_NAME="build-${{ needs.validate.outputs.branch-name-lower }}-${{ needs.validate.outputs.commit-id-short }}"
+          else
+            ARTIFACT_NAME="build-v${{ needs.validate.outputs.version }}"
+          fi
+
+          ARTIFACT_FILE="${ARTIFACT_BASE_DIR}/${ARTIFACT_NAME}.zip"
+          deterministic-zip -D -vr "${ARTIFACT_FILE}" *
+
+          echo "build-version=${ARTIFACT_NAME}" >>"${GITHUB_OUTPUT}"
+          echo "folder=${ARTIFACT_BASE_DIR}" >>"${GITHUB_OUTPUT}"
+          echo "name=${ARTIFACT_NAME}" >>"${GITHUB_OUTPUT}"
+          echo "file=${ARTIFACT_FILE}" >>"${GITHUB_OUTPUT}"
+
+      - name: Compute SHA Hash
+        working-directory: ${{ steps.artifact-release.outputs.folder }}
+        run: |
+          ARTIFACT_NAME="${{ steps.artifact-release.outputs.name }}"
+          sha384sum "${ARTIFACT_NAME}.zip" | tee "${ARTIFACT_NAME}.sha384"
+
+      - name: Upload Artifacts (DevOps GCP Bucket)
```

Comment:
This step will fail if the build artifacts for this commit had been built.

Reply:
Added a step to check if the artifact already existed or not on GS bucket